src/__tests__/index.js

```diff
@@ -316,6 +321,39 @@ test('supports automatic reset of error boundary when resetKeys change', () => {
   expect(handleReset).not.toHaveBeenCalled()
   expect(screen.queryByRole('alert')).not.toBeInTheDocument()
   expect(console.error).not.toHaveBeenCalled()
+
+  // blow it up again
+  userEvent.click(screen.getByText('toggle explode'))
+  screen.getByRole('alert')
+  expect(console.error).toHaveBeenCalledTimes(2)
+  console.error.mockClear()
+
+  // check it calls onResetKeysChange if resetKeys length changes
+  userEvent.click(screen.getByText('toggle extra resetKey'))
+  expect(handleResetKeysChange).toHaveBeenCalledTimes(1)
+  expect(handleResetKeysChange).toHaveBeenCalledWith([true], [true, true])
+  handleResetKeysChange.mockClear()
+  expect(console.error).toHaveBeenCalledTimes(2)
+  console.error.mockClear()
+  screen.getByRole('alert')
+  userEvent.click(screen.getByText('toggle explode'))
+  expect(handleReset).not.toHaveBeenCalled()
+  expect(handleResetKeysChange).toHaveBeenCalledTimes(1)
+  expect(handleResetKeysChange).toHaveBeenCalledWith(
+    [true, true],
+    [false, true],
+  )
+  handleResetKeysChange.mockClear()
+  expect(console.error).toHaveBeenCalledTimes(2)
+  screen.getByRole('alert')
+  console.error.mockClear()
+  userEvent.click(screen.getByText('toggle extra resetKey'))
+  expect(handleReset).not.toHaveBeenCalled()
+  expect(handleResetKeysChange).toHaveBeenCalledTimes(1)
+  expect(handleResetKeysChange).toHaveBeenCalledWith([false, true], [false])
+  handleResetKeysChange.mockClear()
+  expect(screen.queryByRole('alert')).not.toBeInTheDocument()
+  expect(console.error).not.toHaveBeenCalled()
```

Comment:
This is a pretty big block of code with no breaks. Could you add some logical breaks and comments through this to indicate what's happening?